docs/developers/developing_iree/benchmark_suites.md

```diff
@@ -0,0 +1,190 @@
+
+Configure IREE with `-DIREE_BUILD_E2E_TEST_ARTIFACTS=ON`:
+
+```sh
+cmake -GNinja -B "${IREE_BUILD_DIR}" -S "${IREE_REPO}" \
+  -DCMAKE_BUILD_TYPE=RelWithDebInfo \
+  -DCMAKE_C_COMPILER=clang \
+  -DCMAKE_CXX_COMPILER=clang++ \
+  -DIREE_ENABLE_LLD=ON \
+  -DIREE_BUILD_E2E_TEST_ARTIFACTS=ON
+```
+
+Build the benchmark suites and tools:
+
+```sh
+cmake --build "${IREE_BUILD_DIR}" --target \
+  iree-e2e-test-artifacts \
+  iree-benchmark-module
+```
+
+### Run Benchmarks
+
+Export the execution benchmark config:
+
+```sh
+build_tools/benchmarks/export_benchmark_config.py execution > exec_config.json
+```
+
+Run benchmarks (currently only support running on a Linux host):
+
+```sh
+build_tools/benchmarks/run_benchmarks_on_linux.py \
+  --normal_benchmark_tool_dir="${IREE_BUILD_DIR}/tools" \
+  --e2e_test_artifacts_dir="${IREE_BUILD_DIR}/e2e_test_artifacts" \
+  --execution_benchmark_config=exec_config.json \
+  --target_device_name="<target_device_name, e.g. c2-standard-16>" \
+  --output=benchmark_results.json \
+  --verbose \
+  --cpu_uarch="<host CPU uarch, e.g. CascadeLake>"
+# Traces can be collected by adding:
+# --traced_benchmark_tool_dir="${IREE_TRACED_BUILD_DIR}/tools" \
+# --trace_capture_tool=/path/to/iree-tracy-capture \
+# --capture_tarball=captured_tracy_files.tar.gz
+```
+
+Note that: - `<target_device_name>` selects a benchmark group targets a specific
+device: - Common options: - `c2-standard-16` for x86_64 CPU benchmarks -
+`a2-highgpu-1g` for NVIDIA GPU benchmarks - All device names are defined
+[here](/build_tools/python/e2e_test_framework/device_specs) - To run x86_64
+benchmarks, right now `--cpu_uarch` needs to be provided and only `CascadeLake`
+is available currently. - To build traced benchmark tools, see instructions
+[here](/docs/developers/developing_iree/profiling_with_tracy.md).
+
+Filters can be used to select the benchmarks:
+
+```sh
+build_tools/benchmarks/run_benchmarks_on_linux.py \
+  --normal_benchmark_tool_dir="${IREE_BUILD_DIR}/tools" \
+  --e2e_test_artifacts_dir="${IREE_BUILD_DIR}/e2e_test_artifacts" \
+  --execution_benchmark_config=exec_config.json \
+  --target_device_name="c2-standard-16" \
+  --output=benchmark_results.json \
+  --verbose \
+  --cpu_uarch="CascadeLake" \
+  --model_name_regex="MobileBert*" \
+  --driver_filter_regex='local-task' \
+  --mode_regex="4-thread"
+```
+
+### Generate Compilation Statistics (Compilation Benchmarks)
+
+Export the compilation benchmark config:
+
+```sh
+build_tools/benchmarks/export_benchmark_config.py compilation > comp_config.json
+```
+
+Generate the compilation statistics:
+
+```sh
+build_tools/benchmarks/collect_compilation_statistics.py \
+  alpha \
+  --compilation_benchmark_config=comp_config.json \
+  --e2e_test_artifacts_dir="${IREE_BUILD_DIR}/e2e_test_artifacts" \
+  --build_log="${IREE_BUILD_DIR}/.ninja_log" \
+  --output=compile_stats_results.json
+```
+
+Note that you need to use **ninja** (https://ninja-build.org/) to build the
```

Comment:
```suggestion
Note that you need to use `ninja` (https://ninja-build.org/) to build the
```

If it's the name of the executable. Otherwise

```suggestion
Note that you need to use [Ninja](https://ninja-build.org/) to build the
```